### ISSUE TITLE
trezor-core crashes when signing TX with amount > 2^32 satoshi

### ISSUE THREAD

**Issue by karel-3d** [trezor/trezor-core#389]
(I have hit this issue only with BTG, but I think it applies to BTC too)

If I want to sign a BTG transaction with >2^32 satoshis as an output (around 42 btg), trezor-core returns Firmware Error after TxAck.

The error has this stacktrace

```
Traceback (most recent call last):
  File "/Users/karelbilek/dev/trezor-core/src/trezor/wire/__init__.py", line 128, in session_handler
  File "/Users/karelbilek/dev/trezor-core/src/trezor/wire/__init__.py", line 125, in session_handler
  File "/Users/karelbilek/dev/trezor-core/src/trezor/wire/__init__.py", line 163, in protobuf_workflow
  File "/Users/karelbilek/dev/trezor-core/src/trezor/wire/__init__.py", line 150, in protobuf_workflow
  File "/Users/karelbilek/dev/trezor-core/src/trezor/ui/__init__.py", line 125, in inner
  File "/Users/karelbilek/dev/trezor-core/src/trezor/ui/__init__.py", line 122, in inner
  File "/Users/karelbilek/dev/trezor-core/src/apps/wallet/sign_tx/__init__.py", line 38, in sign_tx
  File "/Users/karelbilek/dev/trezor-core/src/apps/wallet/sign_tx/__init__.py", line 28, in sign_tx
  File "/Users/karelbilek/dev/trezor-core/src/apps/wallet/sign_tx/signing.py", line 220, in sign_tx
  File "/Users/karelbilek/dev/trezor-core/src/apps/wallet/sign_tx/signing.py", line 105, in check_tx_fee
  File "/Users/karelbilek/dev/trezor-core/src/apps/wallet/sign_tx/writers.py", line 36, in write_tx_input_check
  File "/Users/karelbilek/dev/trezor-core/src/apps/common/writers.py", line 28, in write_uint32_le
  File "/Users/karelbilek/dev/trezor-core/src/trezor/utils.py", line 40, in ensure
AssertionError:
```

The issue seems to be that trezor-core ensures that the number is <2^32, but satoshi values can be bigger.

**Comment by karel-3d** [comment]
See

https://github.com/trezor/trezor-core/blob/master/src/apps/common/writers.py#L28

https://github.com/trezor/trezor-core/blob/master/src/apps/wallet/sign_tx/writers.py#L36

**Comment by pankrats** [comment]
the same issue reported by a customer https://github.com/satoshilabs/trezor-qa/issues/53

**Comment by karel-3d** [comment]
This issue seems to be that actually the *input* is > 42 btc, not the output, but maybe similar issue will be with outputs?

**Comment by jpochyla** [comment]
Fixed in 5a8388f, thanks!

**Comment by karel-3d** [comment]
yep can confirm
